src/lib/utils/meta_tables/meta_system_information_table.cpp

```diff
@@ -0,0 +1,89 @@
+#ifdef __linux__
+#include <numa.h>
+#include <sys/sysinfo.h>
+#endif
+
+#include <fstream>
+
+#include "meta_system_information_table.hpp"
+
+namespace opossum {
+
+MetaSystemInformationTable::MetaSystemInformationTable()
+    : AbstractMetaSystemTable(TableColumnDefinitions{{"cpu_count", DataType::Int, false},
+                                                     {"system_memory_total_bytes", DataType::Long, false},
+                                                     {"numa_cpu_count", DataType::Int, false},
+                                                     {"cpu_model", DataType::String, false}}) {}
+
+const std::string& MetaSystemInformationTable::name() const {
+  static const auto name = std::string{"system_information"};
+  return name;
+}
+
+std::shared_ptr<Table> MetaSystemInformationTable::_on_generate() const {
+  auto output_table = std::make_shared<Table>(_column_definitions, TableType::Data, std::nullopt, UseMvcc::Yes);
+
+  const auto cpus = _get_cpu_count();
+
+  auto numa_cpus = cpus;
+#ifdef __linux__
+  if (numa_available() != -1) {
+    auto* cpu_mask = numa_allocate_cpumask();
+    numa_sched_getaffinity(0, cpu_mask);
+    numa_cpus = numa_bitmask_weight(cpu_mask);
+    numa_free_cpumask(cpu_mask);
+  }
+#endif
+
+  uint64_t ram;
+#ifdef __linux__
+  struct sysinfo memory_info {};
+  [[maybe_unused]] const auto ret = sysinfo(&memory_info);
+  DebugAssert(ret == 0, "Failed to get sysinfo");

```

Comment:
Could also be a regular Assert, which would save us the maybe_unused. It's not like this is executed over and over again. I'd probably change all DebugAsserts here to Asserts

Reply:
rather 
```
const auto ret = whatever();
Assert(ret);
```
or 
```
Assert(whatever());
```
?

Reply:
The former. First, because it is more consistent with the rest of the codebase, second because it doesn't break if someone decides to make it a DebugAssert again.

Reply:
Generally, assertions should always be completely removable without anything breaking.